lib/phlex/csv.rb

```diff
@@ -48,43 +55,56 @@ def content_type
 
 	def column(header = nil, value)
 		if @_first
-			@_headers << header
+			@_headers << escape(header)
 		elsif header != @_headers[@_current_column_index]
 			raise "Inconsistent header."
 		end
 
-		@_current_row << value
+		@_current_row << escape(value)
 		@_current_column_index += 1
 	end
 
 	def each_item(&block)
 		collection.each(&block)
 	end
 
-	def collection_yielder(record)
+	def yielder(record)
 		yield(record)
 	end
 
 	def template(...)
 		nil
 	end
 
+	# Override and set to `false` to disable rendering headers.
 	def render_headers?
 		true
 	end
 
-	def helpers
-		@_view_context
+	# Override and set to `true` to strip leading and trailing whitespace from values.
+	def trim_whitespace?
+		false
 	end
 
-	def render(renderable)
-		renderable.call(view_context: @_view_context)
+	# Override and set to `false` to disable CSV injection escapes or `true` to enable.
+	def prevent_csv_injection?
+		nil
 	end
 
-	def escape(value)
-		value = value.to_s
+	def helpers
+		@_view_context
+	end
 
-		if value.include?('"') || value.include?(",") || value.include?("\n")
+	def escape(value)
+		value = trim_whitespace? ? value.to_s.strip : value.to_s
+		first_char = value[0]
+		last_char = value[-1]
+
+		if prevent_csv_injection? && FORMULA_PREFIXES[first_char]
+			# Prefix a single quote to prevent Excel, Google Docs, etc. from interpreting the value as a formula.
+			# See https://owasp.org/www-community/attacks/CSV_Injection
+			%("'#{value.gsub('"', '""')}")
+		elsif (!trim_whitespace? && (SPACE_CHARACTERS[first_char] || SPACE_CHARACTERS[last_char])) || value.include?('"') || value.include?(",") || value.include?("\n")
```

Comment:
This line is unfortunately not particularly clear, but it's optimised for performance. For example, we don’t need to check if the first and last characters are whitespace characters if we're trimming whitespace anyway.